src/drivers/uavcannode/UavcanNode.cpp

```diff
@@ -824,6 +824,27 @@ extern "C" int uavcannode_start(int argc, char *argv[])
 		}
 	}
 
+	// Read the static node ID parameter and use it if no valid node_id from shared memory
+	int32_t cannode_node_id = 0;
+	param_get(param_find("CANNODE_NODE_ID"), &cannode_node_id);
+
+	// Check if the static node ID is in range
+	if (cannode_node_id < 0 || cannode_node_id > 127) {
+		PX4_ERR("Invalid static node ID %ld, using dynamic allocation", cannode_node_id);
+		cannode_node_id = 0;
+	}
+
+	// Assign the static node ID if no dynamic allocation is used. Do wen't override a valid node ID from the bootloader?
+	if (node_id == 0 && cannode_node_id > 0 && cannode_node_id <= 127) {
+		node_id = cannode_node_id;
+	}
+
+	// Persist the node ID for the bootloader
+	bootloader_app_shared_t shared_write = {};
+	shared_write.node_id = node_id;
+	shared_write.bus_speed = bitrate;
```

Comment:
There's not really any error checking or validation on the bitrate, it might be safer to do this a bit later after the node has actually started spinning successfully. 

For example you could do it here, and only if there was no dynamic node ID allocation. https://github.com/AlexKlimaj/PX4-Autopilot/blob/a4064072c793787aa9aac149c376bbf3c613d8b0/src/drivers/uavcannode/UavcanNode.cpp#L558-L580